src/js/utils/date-utils.js

```diff
@@ -56,224 +56,215 @@ var DATE_FORMATS = {
     ], function(
         $,
         StringUtils,
-        MomentLib,
-        MomentTZ
+        moment,
+        momentTZ
         ) {
+        var DateUtils, isValid, dateObject, userTimezone, localizeTime, timeDisplay, localize;
+
         /**
-         * Create a formatted, locale-aware string representation of a datetime.
-         *
-         * These functions are meant to act as a general localization function
-         * for UTC datetime objects, creating platform-friendly strings in the user's
-         * local timezone formatted to the user's preferred language
-         *
-         * Most of the heavy lifting is done by the 'moments' js library,
-         * with the moment-with-locales and moment-timezones add-ons
-         *
-         * by sending preferences data, even in an incomplete state, we can generate the appropriate
-         * datetime for surfacing on the frontend
-         *
-         * for example, if you're a French speaker currently accessing the site in Rangoon,
-         * who you have not set your timezone preferences:
-         *
-         * var context = {
-         *    datetime: '2016-10-14 08:00:00+00:00',
-         *    language: "fr",
-         *    }
-         * DateUtils.localize(context)
-         *
-         * which will render: '14 oct. 2016 14:30 MMT'
+         * Helper function to determine variable/context validity.
          *
+         * @param {string} variable to test
+         * @returns {boolean}
          */
-        var DateUtils, DateObject, UserTimeZone, LocalizedTime, TimeDisplay, localize;
+        isValid = function(variable) {
+            var valid = true;
+            if (variable === undefined) {
+                valid = false;
+            } else if (variable === 'undefined') {
+                valid = false;
+            } else if (variable === '') {
+                valid = false;
+            } else if (variable === 'Invalid date') {
+                valid = false;
+            } else {
+                valid = true;
+            }
+            return valid;
+        };
 
         /**
-         * convert sent string to mutable 'moment' datetime object
+         * Attempt to convert a string to mutable 'moment' datetime object.
          *
-         * @param {string} datetime, can be just a date,
-         * or a complex date/time string with TZ offset. This is fairly open, and
-         * many strings will satisfy this (01-01-16 and Jan 1, 2016 will both work)
+         * The input datetimeString can be just a date, a date and time,
+         * or a complex date/time string with TZ offset. This is fairly wide-open, as
+         * many strings will satisfy this requirement.
+         * (e.g. 01-01-16 and Jan 1, 2016, and 2016-01-01 01:00:00+00:00 will work)
          *
-         * @returns {object} a moment.js datetime object in UTC
+         * Recommend using something fully informant, like '2016-01-01 01:00:00+00:00',
```

Comment:
That's fair -- "It is recommended to use a fully informed string, e.g. '2016-01-01 01:00:00+00:00', to limit possible date, time, and timezone ambiguities."